src/lib/logging/LogLayer.js

```diff
@@ -0,0 +1,314 @@
+     * @return {void}
+     */
+    error(...args) {
+        this.printLog(LOG_TYPES.error, args);
+
+        if (!this.boxLoggerExists()) {
+            return;
+        }
+
+        this.loggerRef.error(...args);
+    }
+
+    /**
+     * Interface to the Logger for metric messages.
+     *
+     * @public
+     * @param {string} eventName - Name associated with a specific metric.
+     * @param {*} value - Value of the metric
+     * @return {void}
+     */
+    metric(eventName, value) {
+        this.printLog(LOG_TYPES.metric, [eventName, value]);
+
+        if (!this.boxLoggerExists()) {
+            return;
+        }
+
+        this.loggerRef.metric(eventName, value);
+    }
+
+    /**
+     * Interface for setting File object on the Logger.
+     *
+     * @public
+     * @param {Object} file - The file with file info to attach to logs.
+     * @return {void}
+     */
+    setFile(file) {
+        if (!this.boxLoggerExists()) {
+            return;
+        }
+
+        this.loggerRef.setFile(file);
+    }
+
+    /**
+     * Interface for setting content type on the Logger.
+     *
+     * @public
+     * @param {string} type - Content type for the current logs.
+     * @return {void}
+     */
+    setContentType(type) {
+        if (!this.boxLoggerExists()) {
+            return;
+        }
+
+        this.loggerRef.setContentType(type);
+    }
+
+    /**
+     * Interface for passing through network layer settings to the Logger.
+     *
+     * @public
+     * @param {Object} config - Network related settings to pass to the Logger.
+     * @return {void}
+     */
+    setupNetworkLayer(config) {
+        if (!this.boxLoggerExists()) {
+            return;
+        }
+
+        this.loggerRef.setupNetworkLayer(config);
+    }
+
+    /**
+     * Interface to reset file info for the Logger.
+     *
+     * @public
+     * @return {void}
+     */
+    reset() {
+        if (!this.boxLoggerExists()) {
+            return;
+        }
+
+        this.loggerRef.reset();
+    }
+
+    /**
+     * Save the current state of the logger, for this session, and
+     * clear out any old data.
+     *
+     * @public
+     * @return {void}
+     */
+    save() {
+        if (!this.boxLoggerExists()) {
+            return;
+        }
+
+        // Only saving errors and performance metrics
+        this.loggerRef.save([LOG_TYPES.error, LOG_TYPES.metric]);
+        this.loggerRef.clearCache();
+    }
+
+    //--------------------------------------------------------
+    // PRIVATE
+    //--------------------------------------------------------
+
+    /**
+     * Maps logging type to a logging function in our third-party logger. Defaults
+     * to regular 'info' if nothing available.
+     *
+     * @private
+     * @param {LOG_TYPES|string} type - The log type to lookup a corresponding function on the logger.
+     * @return {Function} A function that can be invoked with a string, to log a message.
```

Comment:
A function that can be invoked with a string, to print a message to the console. Nothing if no correct mapping